RecoTracker/MkFitCore/interface/Track.h

```diff
@@ -0,0 +1,800 @@
+      detachedQuadStep = 24,
+      reservedForUpgrades1 = 25,
+      reservedForUpgrades2 = 26,
+      bTagGhostTracks = 27,
+      beamhalo = 28,
+      gsf = 29,
+      // HLT algo name
+      hltPixel = 30,
+      // steps used by PF
+      hltIter0 = 31,
+      hltIter1 = 32,
+      hltIter2 = 33,
+      hltIter3 = 34,
+      hltIter4 = 35,
+      // steps used by all other objects @HLT
+      hltIterX = 36,
+      // steps used by HI muon regional iterative tracking
+      hiRegitMuInitialStep = 37,
+      hiRegitMuLowPtTripletStep = 38,
+      hiRegitMuPixelPairStep = 39,
+      hiRegitMuDetachedTripletStep = 40,
+      hiRegitMuMixedTripletStep = 41,
+      hiRegitMuPixelLessStep = 42,
+      hiRegitMuTobTecStep = 43,
+      hiRegitMuMuonSeededStepInOut = 44,
+      hiRegitMuMuonSeededStepOutIn = 45,
+      algoSize = 46
+    };
+
+    int algoint() const { return status_.algorithm; }
+    TrackAlgorithm algorithm() const { return TrackAlgorithm(status_.algorithm); }
+    void setAlgorithm(TrackAlgorithm algo) { status_.algorithm = static_cast<unsigned int>(algo); }
+    void setAlgoint(int algo) { status_.algorithm = algo; }
+    // To be used later
+    // bool isStopped() const { return status_.stopped; }
+    // void setStopped()      { status_.stopped = true; }
+
+    static const char* algoint_to_cstr(int algo);
+
+    // ------------------------------------------------------------------------
+
+  protected:
+    TrackState state_;
+    float chi2_ = 0.;
+    float score_ = 0.;
+    short int lastHitIdx_ = -1;
+    short int nFoundHits_ = 0;
+    Status status_;
+    int label_ = -1;
+  };
+
+  //==============================================================================
+  // TrackCand
+  //==============================================================================
+
+  // TrackCand depends on stuff in mkFit/HitStructures, CombCand in particular,
+  // so it is declared / implemented there.
+
+  // class TrackCand : public TrackBase { ... };
+
+  //==============================================================================
+  // Track
+  //==============================================================================
+
+  class Track : public TrackBase {
+  public:
+    Track() {}
+
+    explicit Track(const TrackBase& base) : TrackBase(base) {
+      // Reset hit counters -- caller has to initialize hits.
+      lastHitIdx_ = -1;
+      nFoundHits_ = 0;
+    }
+
+    Track(const TrackState& state, float chi2, int label, int nHits, const HitOnTrack* hits)
+        : TrackBase(state, chi2, label) {
+      reserveHits(nHits);
+      for (int h = 0; h < nHits; ++h) {
+        addHitIdx(hits[h].index, hits[h].layer, 0.0f);
+      }
+    }
+
+    Track(int charge, const SVector3& position, const SVector3& momentum, const SMatrixSym66& errors, float chi2)
+        : TrackBase(charge, position, momentum, errors, chi2) {}
+
+    Track(const Track& t) : TrackBase(t), hitsOnTrk_(t.hitsOnTrk_) {}
+
+    ~Track() {}
```

Comment:
remove

Reply:
or ` = default;`

Reply:
Addressed in Round 2 follow-up PR trackreco/cmssw#76 commit https://github.com/trackreco/cmssw/pull/76/commits/a52b1a9d88304f2becee4962fa1f431c24020fdc